app/components/datafruits-chat-input-message.ts

```diff
@@ -106,25 +122,27 @@ export default class DatafruitsChatInputMessage extends Component {
           );
         }
       },
-      replace: function (shortname) {
+      replace: function (shortname: string) {
         return shortname;
       },
     };
     let usernameComplete = {
       id: 'usernames',
       match: /(^|\s)(\w{2,})$/,
-      search: (term, callback) => {
+      search: (term: string, callback: Function) => {
         let matches;
         matches = Object.keys(this.chat.presences).filter((word) => {
           return word.indexOf(term) === 0 && word !== this.username;
         });
         callback(matches);
       },
-      replace: function (word) {
+      replace: function (word: string) {
         return word + ' ';
       },
     };
-    const editor = new Textarea(document.getElementById('input-message'));
+    let input: HTMLInputElement | null;
+    input = document.getElementById('input-message');
```

Comment:
This fails typechecking because TextareaEditor only supports a textarea element....wonder if we can hack around this or fork the textcomplete package.
https://github.com/yuku/textcomplete